src/block/TNT.php

```diff
@@ -82,6 +83,11 @@ public function onBreak(Item $item, ?Player $player = null, array &$returnedItem
 	}
 
 	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item->getTypeId() === ItemTypeIds::FIRE_CHARGE){
+			$item->pop();
+			$this->ignite();
+			return true;
+		}
```

Comment:
I added a this block, because the fire charge decreases the stack count when used in tnt, but other not Durable items such as enchanted with fire aspect do not.